include/openthread/types.h

```diff
@@ -427,19 +427,46 @@ struct otIp6Address
  */
 typedef struct otIp6Address otIp6Address;
 
-
 /**
  * This structure represents the local and peer IPv6 socket addresses.
  */
 typedef struct otMessageInfo
 {
-    otIp6Address mSockAddr;     ///< The local IPv6 address.
-    otIp6Address mPeerAddr;     ///< The peer IPv6 address.
-    uint16_t     mSockPort;     ///< The local transport-layer port.
-    uint16_t     mPeerPort;     ///< The peer transport-layer port.
-    int8_t       mInterfaceId;  ///< An IPv6 interface identifier.
-    uint8_t      mHopLimit;     ///< The IPv6 Hop Limit.
-    const void  *mLinkInfo;     ///< A pointer to link-specific information.
+    /**
+     * The local IPv6 address.
+     */
+    otIp6Address mSockAddr;
+
+    /**
+     * The peer IPv6 address.
+     */
+    otIp6Address mPeerAddr;
+
+    /**
+     * The local transport-layer port.
+     */
+    uint16_t mSockPort;
+
+    /**
+     * The peer transport-layer port.
+     */
+    uint16_t mPeerPort;
+
+    /**
+     * An IPv6 interface identifier.
+     */
+    int8_t mInterfaceId;
+
+    /**
+     * The IPv6 Hop Limit.
+     */
+    uint8_t mHopLimit;
+
+    /**
+     * A pointer to link-specific information. In case @p mInterfaceId is set to OT_NETIF_INTERFACE_ID_THREAD,
```

Comment:
Should indicate this is only valid for messages received from the Thread radio and is ignored on transmission.